src/react-skycons.js

```diff
@@ -44,6 +44,9 @@ const ReactSkycons = React.createClass({
    this.state.skycons.set(React.findDOMNode(this), Skycons[nextProps.icon]);
   },
 
+  componentWillUnmount: function componentWillUnmount() {
+    this.state.skycons.remove(React.findDOMNode(this));
```

Comment:
seems reasonable if skycons is holding onto references 
does skycons expose a destroy all api or is .remove all they expose?